lightningd/chaintopology.c

```diff
@@ -86,10 +88,22 @@ static void filter_block_txs(struct chain_topology *topo, struct block *b)
 		owned = AMOUNT_SAT(0);
 		txid = b->txids[i];
 		if (txfilter_match(topo->bitcoind->ld->owned_txfilter, tx)) {
```

Comment:
Maybe we should clean this up to return an array of output numbers, rather than scanning multiple times as we do...